source/validate_image.cpp

```diff
@@ -594,6 +594,80 @@ spv_result_t ImagePass(ValidationState_t& _,
   }
 
   switch (opcode) {
+    case SpvOpTypeImage: {
+      assert(result_type == 0);
+
+      ImageTypeInfo info;
+      if (!GetImageTypeInfo(_, inst->words[1], &info)) {
+        return _.diag(SPV_ERROR_INVALID_DATA)
+            << "OpTypeImage: corrupt definition";
+      }
+
+      const SpvOp sampled_type_opcode = _.GetIdOpcode(info.sampled_type);
+      if (sampled_type_opcode != SpvOpTypeVoid &&
+          sampled_type_opcode != SpvOpTypeInt &&
+          sampled_type_opcode != SpvOpTypeFloat) {
+        return _.diag(SPV_ERROR_INVALID_DATA)
+            << "Expected Sampled Type to be either void or numerical scalar "
+            << "type: " << spvOpcodeString(opcode);
+      }
+
+      // Dim is checked elsewhere.
+
+      if (info.depth > 2) {
+        return _.diag(SPV_ERROR_INVALID_DATA)
+            << "Invalid Depth " << info.depth << " (must be 0, 1 or 2): "
```

Comment:
Nit: Hmm, I would typically expect the error message to be something like `invalid Depth parameter for OpTypeImage: <the-wrong-number>`, or `OpTypeImage: invalid Depth parameter: <the-wrong-number>`. The current error message reads like we are providing `TypeImage` as the `Depth` parameter.